user-interface/src/main/java/life/qbic/datamanager/parser/MetadataConverter.java

```diff
@@ -0,0 +1,352 @@
+ *
+ * <ul>
+ *   <li>Proteomics Measurement {@link ProteomicsMeasurementProperty}</li>
+ *   <li>NGS Measurement {@link NGSMeasurementProperty}</li>
+ * </ul>
+ *
+ * @since 1.4.0
+ */
+public class MetadataConverter implements MeasurementMetadataConverter {
+
+  private static final Logger log = logger(MetadataConverter.class);
+
+  private MetadataConverter() {
+  }
+
+  public static MetadataConverter measurementConverter() {
+    return new MetadataConverter();
+  }
+
+  /**
+   * Generates a hit map, storing the number of matches of a defined set of String values (hit
+   * values), in a target of interest collection of String values.
+   * <p>
+   * The resulting map will contain the number of occurrences of every value in the hit values
+   * collection found in the target collection to investigate.
+   *
+   * @param target    the collection of interest to search in
+   * @param hitValues a set of distinct values, that should be represented in the hit result map
+   * @return a hit result map, containing the number of occurrences of every hit value in the target
+   * String collection (0, if no target was found for a value).
+   * @since 1.4.0
+   */
+  private static Map<String, Integer> countHits(Collection<String> target, Set<String> hitValues,
+      String... ignoredProperties) {
+    Map<String, Integer> hits = new HashMap<>();
+    for (String t : hitValues) {
+      hits.put(t, 0);
+    }
+    for (String s : target) {
+      if (hitValues.contains(s)) {
+        var currentHit = hits.get(s);
+        hits.put(s, ++currentHit);
+      }
+    }
+    for (String ignoredProperty : ignoredProperties) {
+      if (hits.containsKey((ignoredProperty))) {
+        hits.remove(ignoredProperty);
+      }
+      ;
+    }
+    return hits;
+  }
+
+  @Override
+  public List<MeasurementMetadata> convert(ParsingResult parsingResult, boolean ignoreMeasurementId)
+      throws UnknownMetadataTypeException {
+    Objects.requireNonNull(parsingResult);
+    var properties = parsingResult.keys().keySet();
+    if (looksLikeNgsMeasurement(properties, ignoreMeasurementId)) {
+      return convertNGSMeasurement(parsingResult);
+    } else if (looksLikeProteomicsMeasurement(properties, ignoreMeasurementId)) {
+      return convertProteomicsMeasurement(parsingResult);
+    } else {
+      throw new UnknownMetadataTypeException(
+          "Unknown metadata type: cannot match properties to any known metadata type. Provided [%s]".formatted(
+              String.join(", ", properties)));
+    }
+  }
+
+  private List<MeasurementMetadata> convertProteomicsMeasurement(ParsingResult parsingResult) {
+    var result = new ArrayList<MeasurementMetadata>();
+    var keyIndices = parsingResult.keys();
+    for (List<String> valueEntry : parsingResult.values()) {
+      var pxpMetaDatum = new ProteomicsMeasurementMetadata(
+          safeListAccess(valueEntry, keyIndices.getOrDefault(MEASUREMENT_ID.propertyName(), -1),
```

Comment:
maybe add a little javadoc to indicate that the value `-1` is present for missing columns?